File: frontend/src/api/calendar.ts
import { api } from './index'

export interface CalendarEvent {
  event_id: number
  content_id: number
  platform: string
  scheduled_date: string
  status: string
  content_title?: string
  content_type?: string
}

export async function getEvents(days = 30) {
  const { data } = await api.get<CalendarEvent[]>('/calendar/events', { params: { days } })
  return data
}

export async function createEvent(payload: { content_id: number; platform: string; scheduled_date: string }) {
  const { data } = await api.post<{ event_id: number }>('/calendar/events', payload)
  return data
}

File: src/api/schemas/calendar.py
from datetime import date
from typing import Optional

from pydantic import BaseModel


class CalendarEventCreate(BaseModel):
    content_id: int
    platform: str
    scheduled_date: date


class CalendarEventResponse(BaseModel):
    event_id: int
    content_id: int
    platform: str
    scheduled_date: str
    status: str
    content_title: Optional[str] = None
    content_type: Optional[str] = None
